src/slptokens/token.h

```diff
@@ -0,0 +1,134 @@
+    SLP_TX_TYPE tx_type; // 4-6 bytes
+
+    // NOT IN GENESIS
+    std::vector<uint8_t> token_id; // 32 bytes, token genesis transaction hash
+
+    // NOT IN SEND
+    uint8_t mint_baton_vout; //(0 bytes or 1 byte between 0x02-0xff), if 0 there is no mint baton
+
+    // ONLY IN SEND
+    std::vector<uint64_t> token_output_quantitys; // this can be a max size of 19
+
+    // ONLY IN GENESIS
+    std::vector<uint8_t> token_ticker; // (0 or more bytes, suggested utf-8)
+    std::vector<uint8_t> token_name; // (0 or more bytes, suggested utf-8)
+    std::vector<uint8_t> token_document_url; // (0 or more bytes, suggested ascii)
+    std::vector<uint8_t> token_document_hash; // 32 bytes
+    uint8_t decimals;
+    uint64_t initial_token_mint_quantity;
+    // ONLY IN SEND
+    uint64_t additional_token_quantity;
+
+public:
+    // UTXO Fields
+    int nHeight;
+
+private:
+    bool DetermineDynamicSize(const CScript &scriptPubKeyIn, size_t &index, uint32_t &return_size);
+    SLP_TX_TYPE ParseType(const CScript &scriptPubKeyIn, size_t &index, const uint32_t &tx_type_size);
+    uint8_t ParseBytesGenesis(const CScript &scriptPubKeyIn, size_t &index);
+    uint8_t ParseBytesMint(const CScript &scriptPubKeyIn, size_t &index);
+    uint8_t ParseBytesSend(const CScript &scriptPubKeyIn, size_t &index);
+    uint8_t ParseBytesCommit(const CScript &scriptPubKeyIn, size_t &index);
+    uint8_t _ParseBytes(const CScript &scriptPubKeyIn);
+
+public:
+    CSLPToken(const int &nHeightIn) { nHeight = nHeightIn; }
+    CSLPToken() { SetNull(); }
+    void SetNull()
+    {
+        token_type = 0;
+        tx_type = SLP_NULL;
+        token_id.clear();
+        mint_baton_vout = 0;
+        token_output_quantitys = std::vector<uint64_t>(19, 0); // prefill with 19 0's
+        token_ticker.clear();
+        token_name.clear();
+        token_document_url.clear();
+        token_document_hash.clear();
+        decimals = 0;
+        initial_token_mint_quantity = 0;
+        additional_token_quantity = 0;
+        nHeight = 0;
+    }
+
+    ADD_SERIALIZE_METHODS;
+    // this is not that great, it might be better to make a token header
+    // and then different types that extend the header to avoid always having all fields
+    // in 1 token class like this
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream &s, Operation ser_action)
+    {
+        uint8_t tx_type_ser;
+        if (!ser_action.ForRead()) // Write
+        {
+            tx_type_ser = (uint8_t)tx_type;
+        }
+        READWRITE(token_type);
+        READWRITE(tx_type_ser);
+        READWRITE(token_id);
+        READWRITE(mint_baton_vout);
+        READWRITE(token_output_quantitys);
+        READWRITE(token_ticker);
+        READWRITE(token_name);
+        READWRITE(token_document_url);
+        READWRITE(token_document_hash);
+        READWRITE(decimals);
+        READWRITE(initial_token_mint_quantity);
+        READWRITE(additional_token_quantity);
```

Comment:
nHeight is not being serialized.  In a comment near the nHeight declaration, please make it clear that this field is RAM-only and why.  you'll see that other structures do the same thing, although weak on the "why"

Reply:
i have opted to remove nHeight instead. it was being used to track the last time the slpDB was updated similar in the way that the utxo db uses the block height to determine the last time it was updated. the problem with this is that the slp db is not garenteed to be updated every block like the utxodb is so to even verify it was correct you would need to rescan the blocks until that height. it is better to just use the utxo bestknownblock to keep track of the last time both were (possibly) altered.